### PR TITLE
Add vscode editor integration

### LINKED ISSUE #123: Document language server setup
Now that we have a LSP implementation with #108 , we need to document how to configure it for the various editors that support language servers (e.g. VSCode, Vim, and Emacs).

This also requires creating the missing editor plugin to handle swarm-lang (e.g. associate to file extension and provide syntax highlighting).

Here is an example of such documentation: https://github.com/dhall-lang/dhall-lang/blob/master/docs/howtos/Text-Editor-Configuration.md

### PR DESCRIPTION
**Pull request by xsebek**
- adds `editors` directory
- migrates [xsebek/swarm-vscode](https://github.com/xsebek/swarm-vscode) repo to `editors/`
- updates the highlighting (drill, sum types,...)
- adds some notes how to test and build the extension
- part of #100 
- closes #123

### CODE REVIEW

**Comment by xsebek** [comment]
@byorgey I wanted some logo in the extension list, so quickly hacked together this one:

Screenshot from 2021-11-11 16-05-39

**Comment by xsebek** [comment]
As an aside, I wonder why the CI is running - Mergify correctly found that it is not needed :confused:

**Comment by byorgey** [comment]
Looks nice! =D

**Comment by xsebek** [comment]
Nice, it seems that #221 is working! :partying_face:

**Comment by xsebek** [comment]
*Version 0.0.3 is alive!* :zombie: 

Screenshot from 2021-11-11 16-44-49

https://marketplace.visualstudio.com/items?itemName=xsebek.swarm-language

File: editors/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+the `M-x lsp` service in a *swarm-mode* buffer.
+
+## VS Code and VS Codium
+
+The [swarm-lang](./vscode) extension provides highlighting and an
+LSP client. That is if you have `swarm` executable in PATH, then
+the executable will be used as LSP server to show errors as you type.
+
+You can get it by:
+- installing from MS marketplace ([link](XXX))
```
Comment (byorgey) [inline editors/README.md:27]: Can this be filled in now?

**Review by byorgey — APPROVED** [review]
LGTM, though I didn't really look at any of the VSCode specific stuff.

File: editors/README.md (hunk 1 above)
Comment (xsebek) [reply editors/README.md]: Yes, I forgot to add the link :slightly_smiling_face:

**Pull request merged**